### ISSUE TITLE
Tests not executing when using newman programmatically 

### ISSUE THREAD

**Issue by jonniedarko** [postmanlabs/newman#610]
If I try to run my postman collection tests using the following code it does not fail but does not execute:
```js
var newman = require('newman'); // require newman in your project

newman.run({
    collection: require('./tests/api.postman_collection.json'),
    reporters: 'cli'
}, function (err) {
    if (err) { throw err; }
    console.log('collection run complete!');
});

```

this produces the output:
```bash
newman

api-server-pm-tests

┌─────────────────────────┬──────────┬──────────┐
│                         │ executed │   failed │
├─────────────────────────┼──────────┼──────────┤
│              iterations │        0 │        0 │
├─────────────────────────┼──────────┼──────────┤
│                requests │        0 │        0 │
├─────────────────────────┼──────────┼──────────┤
│            test-scripts │        0 │        0 │
├─────────────────────────┼──────────┼──────────┤
│      prerequest-scripts │        0 │        0 │
├─────────────────────────┼──────────┼──────────┤
│              assertions │        0 │        0 │
├─────────────────────────┴──────────┴──────────┤
│ total run duration: 4ms                       │
├───────────────────────────────────────────────┤
│ total data received: 0B (approx)              │
├───────────────────────────────────────────────┤
│ average response time: 0ms                    │
└───────────────────────────────────────────────┘
collection run complete!

```

But if I just run using the globally install newman command
```
newman run /pat/to/tests/api.postman_collection.json
```
I get the expected output:

```bash
newman

api-server-pm-tests

→ http://api.local.dev/xyz
  GET http://api.local.dev/xyz [200 OK, 385B, 750ms]
  ✓  Status code is 200
  ✓  No Error
  ✓  Correct Environment
  ✓  No Image

┌─────────────────────────┬──────────┬──────────┐
│                         │ executed │   failed │
├─────────────────────────┼──────────┼──────────┤
│              iterations │        1 │        0 │
├─────────────────────────┼──────────┼──────────┤
│                requests │        1 │        0 │
├─────────────────────────┼──────────┼──────────┤
│            test-scripts │        1 │        0 │
├─────────────────────────┼──────────┼──────────┤
│      prerequest-scripts │        0 │        0 │
├─────────────────────────┼──────────┼──────────┤
│              assertions │        4 │        0 │
├─────────────────────────┴──────────┴──────────┤
│ total run duration: 1093ms                    │
├───────────────────────────────────────────────┤
│ total data received: 91B (approx)             │
├───────────────────────────────────────────────┤
│ average response time: 750ms                  │
└───────────────────────────────────────────────┘
```

The file used in both cases is the exact same and was exported from postman (using Collection v1)

I have verified this behaviour on both `node v6.4.0` & `node v4.5.0`

**Comment by kunagpal** [comment]
Hey, @jonniedarko I'm unable to reproduce this issue on my end. Would it be possible for you to share the collection file, sans the sensitive details?

**Comment by jonniedarko** [comment]
@kunagpal Sure, here it is:
```json
{
	"id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
	"name": "api-pm-tests",
	"requests": [
		{
			"id": "yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy",
			"headers": "Authorization: Hawk id=\"123456789\", ts=\"1473793108\", nonce=\"EZwQi1\", mac=\"X123Y45671sP2bnZgj/ixCzCNE22Qvu/uyZTQWbV18=\"\n",
			"url": "http://api.local.dev/xyz",
			"pathVariables": {},
			"preRequestScript": null,
			"method": "GET",
			"collectionId": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
			"data": null,
			"dataMode": "params",
			"name": "http://api.local.dev/xyz",
			"description": "",
			"descriptionFormat": "html",
			"time": 1473793139780,
			"version": 2,
			"responses": [],
			"tests": null,
			"currentHelper": "normal",
			"helperAttributes": {}
		}
	],
	"order": [
		"yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy",
	],
	"timestamp": 1473793139782
}
```

**Comment by kunagpal** [comment]
@jonniedarko Thanks for reporting this bug, we'll be adding a fix for the same.

**Comment by kunagpal** [comment]
@jonniedarko  This bug should have been fixed in Newman `v3.1.1`, please check it out and get back to us if your issue persists. Thanks!

**Comment by jonniedarko** [comment]
👍
